graviton/blackbox.py

```diff
@@ -303,6 +301,11 @@ def asdict(self, drop_nones: bool = True) -> Dict[str, Any]:
 
         return {k: v for k, v in d.items() if v is not None}
 
+    def update(self, other: "DryRunTransactionParams") -> None:
+        for k, v in vars(other).items():
+            if v is not None:
+                setattr(self, k, v)
+
```

Comment:
Hmmm... I do believe this is a reference in 634:

https://github.com/algorand/pyteal/blob/2dafd5213adfcca4ad97aebf06f0ca7002c17d85/tests/blackbox.py#L767

Reply:
I see, so we do need to keep this method. A few other suggestions here:
- a bit worried about using `vars`, probably use https://stackoverflow.com/a/70168093 ?
- suggest renaming as `update_against_params`, for we are taking non empty fields in another instance and apply to self.

Reply:
* Regarding first bullet: this is a good point. But since I need both keys and values, I'll use the `asdict()` method of `dataclasses` instead
* Regarding second bullet: how about `update_fields` ?

Reply:
sounds good to me!

Reply:
Addressed in https://github.com/algorand/graviton/pull/49/commits/d8541f210844dd34c6ffbe7d2533b7945c1b9fe8 + https://github.com/algorand/graviton/commit/809d54466f14bb7c67cd5989b6b9dbec35dbcfc5